Malformed if-match value triggers exception

### Steps to reproduce:

1. Binding config
```
"http_kafka_proxy0":

        {
            "type" : "http-kafka",
            "kind": "proxy",
            "routes":
            [
                {
                    "when":
                    [
                        {
                            "method": "PUT",
                            "path": "/items/{id}"
                        },
                       "with":
                      {
                          "capability": "produce",
                          "topic": "items-commands",
                          "key": "${params.id}",
                          "reply-to": "item-replies"
                      }
                        .....
```
2. Curl command:

```
curl -v \
    -X "PUT" "http://localhost:8080/items/9417E83E-313E-468E-AC7C-1BCE0BAF9712" \
    -H "Idempotency-Key: `uuidgen`" \
    -H "Content-Type: application/json" \
    -H "If-Match: wrong" \
    -d "{\"@type\": \"UpdateItemCommand\", \"description\": \"Test1\"}"
```

**Exception:**

```
org.agrona.concurrent.AgentTerminationException: java.lang.IllegalArgumentException: negative length: -1

at io.aklivity.zilla.runtime.engine@0.9.17/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.doWork(DispatchAgent.java:572)

at org.agrona.core/org.agrona.concurrent.AgentRunner.doDutyCycle(AgentRunner.java:291)

at org.agrona.core/org.agrona.concurrent.AgentRunner.run(AgentRunner.java:164)

at java.base/java.lang.Thread.run(Thread.java:833)

Caused by: java.lang.IllegalArgumentException: negative length: -1

at org.agrona.core/org.agrona.concurrent.UnsafeBuffer.lengthCheck(UnsafeBuffer.java:1705)

at org.agrona.core/org.agrona.concurrent.UnsafeBuffer.putBytes(UnsafeBuffer.java:944)

at io.aklivity.zilla.runtime.binding.http.kafka@0.9.17/io.aklivity.zilla.runtime.binding.http.kafka.internal.types.OctetsFW$Builder.set(OctetsFW.java:77)

at io.aklivity.zilla.runtime.binding.http.kafka@0.9.17/io.aklivity.zilla.runtime.binding.http.kafka.internal.types.KafkaHeaderFW$Builder.value(KafkaHeaderFW.java:256)

at io.aklivity.zilla.runtime.binding.http.kafka@0.9.17/io.aklivity.zilla.runtime.binding.http.kafka.internal.config.HttpKafkaWithProduceResult.lambda$header$3(HttpKafkaWithProduceResult.java:168)

at io.aklivity.zilla.runtime.binding.http.kafka@0.9.17/io.aklivity.zilla.runtime.binding.http.kafka.internal.types.Array32FW$Builder.item(Array32FW.java:192)

at io.aklivity.zilla.runtime.binding.http.kafka@0.9.17/io.aklivity.zilla.runtime.binding.http.kafka.internal.config.HttpKafkaWithProduceResult.header(HttpKafkaWithProduceResult.java:164)

at io.aklivity.zilla.runtime.binding.http.kafka@0.9.17/io.aklivity.zilla.runtime.binding.http.kafka.internal.config.HttpKafkaWithProduceResult.lambda$headers$1(HttpKafkaWithProduceResult.java:131)

at io.aklivity.zilla.runtime.binding.http.kafka@0.9.17/io.aklivity.zilla.runtime.binding.http.kafka.internal.types.Array32FW.forEach(Array32FW.java:68)

at io.aklivity.zilla.runtime.binding.http.kafka@0.9.17/io.aklivity.zilla.runtime.binding.http.kafka.internal.config.HttpKafkaWithProduceResult.headers(HttpKafkaWithProduceResult.java:131)

at io.aklivity.zilla.runtime.binding.http.kafka@0.9.17/io.aklivity.zilla.runtime.binding.http.kafka.internal.stream.HttpKafkaProxyFactory$HttpProduceSyncProxy.lambda$onHttpBegin$2(HttpKafkaProxyFactory.java:3179)

at io.aklivity.zilla.runtime.binding.http.kafka@0.9.17/io.aklivity.zilla.runtime.binding.http.kafka.internal.types.stream.KafkaMergedDataExFW$Builder.headers(KafkaMergedDataExFW.java:300)

at io.aklivity.zilla.runtime.binding.http.kafka@0.9.17/io.aklivity.zilla.runtime.binding.http.kafka.internal.stream.HttpKafkaProxyFactory$HttpProduceSyncProxy.lambda$onHttpBegin$3(HttpKafkaProxyFactory.java:3179)

at io.aklivity.zilla.runtime.binding.http.kafka@0.9.17/io.aklivity.zilla.runtime.binding.http.kafka.internal.types.stream.KafkaDataExFW$Builder.merged(KafkaDataExFW.java:265)

at io.aklivity.zilla.runtime.binding.http.kafka@0.9.17/io.aklivity.zilla.runtime.binding.http.kafka.internal.stream.HttpKafkaProxyFactory$HttpProduceSyncProxy.onHttpBegin(HttpKafkaProxyFactory.java:3174)

at io.aklivity.zilla.runtime.binding.http.kafka@0.9.17/io.aklivity.zilla.runtime.binding.http.kafka.internal.stream.HttpKafkaProxyFactory$HttpProduceSyncProxy.onHttpMessage(HttpKafkaProxyFactory.java:3106)

at io.aklivity.zilla.runtime.binding.http.kafka@0.9.17/io.aklivity.zilla.runtime.binding.http.kafka.internal.stream.HttpKafkaProxyFactory.lambda$newStream$4(HttpKafkaProxyFactory.java:224)

at io.aklivity.zilla.runtime.engine@0.9.17/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.handleDefaultReadInitial(DispatchAgent.java:1073)

at io.aklivity.zilla.runtime.engine@0.9.17/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.handleReadInitial(DispatchAgent.java:1010)

at io.aklivity.zilla.runtime.engine@0.9.17/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.handleRead(DispatchAgent.java:953)

at io.aklivity.zilla.runtime.engine@0.9.17/io.aklivity.zilla.runtime.engine.internal.concurent.ManyToOneRingBuffer.read(ManyToOneRingBuffer.java:181)

at io.aklivity.zilla.runtime.engine@0.9.17/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.doWork(DispatchAgent.java:566)
```

--- a/runtime/binding-http-kafka/src/main/java/io/aklivity/zilla/runtime/binding/http/kafka/internal/config/HttpKafkaWithProduceResult.java
+++ b/runtime/binding-http-kafka/src/main/java/io/aklivity/zilla/runtime/binding/http/kafka/internal/config/HttpKafkaWithProduceResult.java
@@ -164,14 +164,17 @@ public class HttpKafkaWithProduceResult
         {
             final String16FW value = HTTP_HEADER_NAME_IF_MATCH.equals(name) ? ifMatch : header.value();
 
-            builder.item(i -> i
-                .nameLen(name.length())
-                .name(name.value(), 0, name.length())
-                .valueLen(value.length())
-                .value(value.value(), 0, value.length()));
+            if (value != null)
+            {
+                builder.item(i -> i
+                    .nameLen(name.length())
+                    .name(name.value(), 0, name.length())
+                    .valueLen(value.length())
+                    .value(value.value(), 0, value.length()));
 
-            hash.updateHash(name.value());
-            hash.updateHash(value.value());
+                hash.updateHash(name.value());
+                hash.updateHash(value.value());
+            }
         }
     }
 

--- a/runtime/binding-http-kafka/src/main/java/io/aklivity/zilla/runtime/binding/http/kafka/internal/config/HttpKafkaWithResolver.java
+++ b/runtime/binding-http-kafka/src/main/java/io/aklivity/zilla/runtime/binding/http/kafka/internal/config/HttpKafkaWithResolver.java
@@ -382,7 +382,11 @@ public final class HttpKafkaWithResolver
         final HttpHeaderFW httpIfMatch = httpHeaders.matchFirst(h -> HEADER_NAME_IF_MATCH.equals(h.name()));
         if (httpIfMatch != null && etagMatcher.reset(httpIfMatch.value().asString()).matches())
         {
-            ifMatch = new String16FW(etagMatcher.group("ifmatch"));
+            final String group = etagMatcher.group("ifmatch");
+            if (group != null)
+            {
+                ifMatch = new String16FW(group);
+            }
         }
 
         return new HttpKafkaWithProduceResult(

--- a/runtime/binding-http-kafka/src/test/java/io/aklivity/zilla/runtime/binding/http/kafka/internal/stream/HttpKafkaProxyIT.java
+++ b/runtime/binding-http-kafka/src/test/java/io/aklivity/zilla/runtime/binding/http/kafka/internal/stream/HttpKafkaProxyIT.java
@@ -103,6 +103,16 @@ public class HttpKafkaProxyIT
     @Test
     @Configuration("proxy.delete.item.json")
     @Specification({
+        "${http}/delete.item.if.match.no.etag/client",
+        "${kafka}/delete.item/server"})
+    public void shouldDeleteItemIfMatchNoEtag() throws Exception
+    {
+        k3po.finish();
+    }
+
+    @Test
+    @Configuration("proxy.delete.item.json")
+    @Specification({
         "${http}/delete.item.read.abort/client",
         "${kafka}/delete.item.read.abort/server"})
     public void shouldNotDeleteItemWhenReadAbort() throws Exception

--- a/specs/binding-http-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/http/kafka/streams/http/delete.item.if.match.no.etag/client.rpt
+++ b/specs/binding-http-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/http/kafka/streams/http/delete.item.if.match.no.etag/client.rpt
@@ -1,0 +1,40 @@
+#
+# Copyright 2021-2022 Aklivity Inc
+#
+# Licensed under the Aklivity Community License (the "License"); you may not use
+# this file except in compliance with the License.  You may obtain a copy of the
+# License at
+#
+#   https://www.aklivity.io/aklivity-community-license/
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations under the License.
+#
+
+connect "zilla://streams/http0"
+        option zilla:window 8192
+        option zilla:transmission "half-duplex"
+        option zilla:update "proactive"
+
+write zilla:begin.ext ${http:beginEx()
+                            .typeId(zilla:id("http"))
+                            .header(":method", "DELETE")
+                            .header(":scheme", "https")
+                            .header(":authority", "example.com:9090")
+                            .header(":path", "/items/92d0bf92-63e0-4cfc-ae73-71dee92d1544")
+                            .header("if-match", "AQQABAIC")
+                            .header("idempotency-key", "7101a8ea-5745-43c8-8d0c-b2f9baba7004")
+                            .build()}
+
+connected
+
+write close
+
+read zilla:begin.ext ${http:matchBeginEx()
+                           .typeId(zilla:id("http"))
+                           .header(":status", "204")
+                           .build()}
+
+read closed

--- a/specs/binding-http-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/http/kafka/streams/http/delete.item.if.match.no.etag/server.rpt
+++ b/specs/binding-http-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/http/kafka/streams/http/delete.item.if.match.no.etag/server.rpt
@@ -1,0 +1,40 @@
+#
+# Copyright 2021-2022 Aklivity Inc
+#
+# Licensed under the Aklivity Community License (the "License"); you may not use
+# this file except in compliance with the License.  You may obtain a copy of the
+# License at
+#
+#   https://www.aklivity.io/aklivity-community-license/
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations under the License.
+#
+
+accept "zilla://streams/http0"
+       option zilla:window 8192
+       option zilla:transmission "half-duplex"
+accepted
+
+read zilla:begin.ext ${http:matchBeginEx()
+                           .typeId(zilla:id("http"))
+                           .header(":method", "DELETE")
+                           .header(":path", "/items/92d0bf92-63e0-4cfc-ae73-71dee92d1544")
+                           .header("if-match", "AQQABAIC")
+                           .header("idempotency-key", "7101a8ea-5745-43c8-8d0c-b2f9baba7004")
+                           .build()}
+
+connected
+
+read closed
+
+write zilla:begin.ext ${http:beginEx()
+                            .typeId(zilla:id("http"))
+                            .header(":status", "204")
+                            .build()}
+
+write flush
+
+write close

--- a/specs/binding-http-kafka.spec/src/test/java/io/aklivity/zilla/specs/binding/http/kafka/streams/HttpIT.java
+++ b/specs/binding-http-kafka.spec/src/test/java/io/aklivity/zilla/specs/binding/http/kafka/streams/HttpIT.java
@@ -64,6 +64,15 @@ public class HttpIT
 
     @Test
     @Specification({
+        "${http}/delete.item.if.match.no.etag/client",
+        "${http}/delete.item.if.match.no.etag/server"})
+    public void shouldNotDeleteItemIfMatchNoEtag() throws Exception
+    {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({
         "${http}/delete.item.read.abort/client",
         "${http}/delete.item.read.abort/server"})
     public void shouldNotDeleteItemWhenReadAbort() throws Exception

